pkg/stub/handler.go

```diff
@@ -497,18 +651,23 @@ func (h *Handler) processFiles(dir string, files []os.FileInfo, opcfg *v1alpha1.
 			if err != nil {
 				return h.processError(opcfg, v1alpha1.SamplesExist, corev1.ConditionUnknown, err, "%v error reading file %s", dir+"/"+file.Name())
 			}
+			is, err := h.imageclientwrapper.Get("openshift", imagestream.Name, metav1.GetOptions{})
+			if err != nil && !kerrors.IsNotFound(err) {
+				return h.processError(opcfg, v1alpha1.SamplesExist, corev1.ConditionUnknown, err, "unexpected imagestream get error: %v")
+			}
+
+			if kerrors.IsNotFound(err) {
+				// testing showed that we get an empty is vs. nil in this case
+				is = nil
+			}
 
 			if _, isok := h.skippedImagestreams[imagestream.Name]; !isok {
 				h.updateDockerPullSpec([]string{"docker.io", "registry.redhat.io", "registry.access.redhat.com", "quay.io"}, imagestream, opcfg)
-				is, err := h.imageclientwrapper.Get("openshift", imagestream.Name, metav1.GetOptions{})
-				if err != nil && !kerrors.IsNotFound(err) {
-					return h.processError(opcfg, v1alpha1.SamplesExist, corev1.ConditionUnknown, err, "unexpected imagestream get error: %v")
-				}
 
-				if kerrors.IsNotFound(err) {
-					// testing showed that we get an empty is vs. nil in this case
-					is = nil
+				if imagestream.Labels == nil {
+					imagestream.Labels = map[string]string{}
 				}
+				imagestream.Labels[v1alpha1.SamplesResourceLabel] = ""
```

Comment:
should set it to something, e.g. "true"

Reply:
the label not being set, and the label being set to false(or anything that isn't "true"), would both have the same meaning for the operator, effectively.

Setting it to true allows us to set it to false in the future if we want to (or just remove it as you're doing today, which is also fine).

So i think i'd rather see it explicitly set to false (And the selector to select on =true, per your (c)) so we don't paint ourselves into a corner w/ respect to how the label is used in the future.